nemo/utils/callbacks/nemo_model_checkpoint.py

```diff
@@ -228,6 +228,8 @@ def on_save_checkpoint(self, trainer, pl_module, checkpoint):
             maybe_injected_best_model_path = self.best_model_path
 
         if self.save_best_model:
+            if torch.distributed.is_initialized():
```

Comment:
Testing this key thing. I have what looks like a race condition issue now with a GPT model I'm training.